psi4/src/psi4/dlpno/sparse.cc

```diff
@@ -64,6 +64,27 @@ std::vector<int> merge_lists(const std::vector<int> &l1, const std::vector<int>
 
 }
 
+std::vector<int> index_list(const std::vector<int> &l1, const std::vector<int> &l2) {
```

Comment:
maybe `indices_from_sorted_subset` or something since you're not checking the assumptions.